### PR TITLE
feat: Support selinux_modules during bootc builds

### PR DESCRIPTION
**Pull request by martinpitt**
Feature: Support configuring modules during container builds.

Reason: This is particularly useful for building bootc derivative OSes.

Result: The `selinux_modules` option is now officially supported for
calling during a bootc container build.

This was already mostly working, we just need to get past the current SELinux
status check. Disable that check when the system is not booted. Enable
tests_selinux_modules, and add an SELinux add/disable check to tests_bootc_e2e.

https://issues.redhat.com/browse/RHEL-95404

### CODE REVIEW

File: library/local_semodule.py [hunk 1]
```diff
@@ -43,6 +43,12 @@
       - Use other than current SELinux module store, e.g. mls, minimum, refpolicy ...
     type: str
     default: ""
+  booted:
```
[the remark this answers is not in the document]
Comment (richm) [reply library/local_semodule.py:46]: `booted` here has a very different semantic meaning than `reload`

**Comment by richm** [comment]
[citest_bad]

File: library/local_semodule.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bachradsusi) [reply library/local_semodule.py:46]: From my POV, the difference between booted and not booted is similar to the state when you set `store` to something else than currently running policy type - you can install a module, you can rebuild the policy but you can't load it. The documentation for `booted` also mentions that the difference is that it does not reload the policy. So if the module has option `reload` it's clear that it would not reload policy. OTOH using `booted` for cases when a user does not need to reload the policy sounds strange. 

Also it's not necessary to check runtime status if the reload is not requested - it does not make sense. Same would apply to using this module with `store: mls` option in system with `targeted` policy.

[the remark this answers is not in the document]
Comment (bachradsusi) [reply library/local_semodule.py:46]: I'd say that the current (before this patch) implementation is not good in this sense and should be improved if there's such requirement as this one.

[the remark this answers is not in the document]
Comment (richm) [reply library/local_semodule.py:46]: ok - I guess @martinpitt check out the community.general.sefcontext and community.general.selogin modules for how the `reload` parameter is used - if it has the same behavior as you are proposing here for the `booted` parameter, then use `reload` instead of `booted` for local_semodule.py

[the remark this answers is not in the document]
Comment (martinpitt) [reply library/local_semodule.py:46]: @bachradsusi In case it's unclear: this option is *not* user-visible API. It's just an internal helper module from the role. As a compromise, how about I call it `_check_active` ? Because that's exactly what this flag does. I really don't want to name it "reload" -- that's not at all what it's doing.

[the remark this answers is not in the document]
Comment (martinpitt) [reply library/local_semodule.py:46]: Renamed to `_check_active` and updated the documentation. If we want to make the role actually reload (on booted systems), I propose to do that in a separate PR, as that's unrelated.

**Comment by martinpitt** [comment]
[citest]

**Comment by bachradsusi** [comment]
The problem is not specific to bootc, but to issue in local_semodule.py does not work in selinux disabled environment.
local_semodule.py should follow functionality of `semanage module -a module.pp` and this works in SELinux disabled systems (if there's /etc /selinux/config and policy installed).

```
diff --git a/library/local_semodule.py b/library/local_semodule.py
index 5f2c7195ebb5..b272d0cff41b 100644
--- a/library/local_semodule.py
+++ b/library/local_semodule.py
@@ -356,9 +356,6 @@ def main():
             exception=SEMANAGE_IMP_ERR,
         )
 
-    if not get_runtime_status():
-        module.fail_json(msg="SELinux is disabled on this host.")
-
     path = module.params["path"]
     name = module.params["name"]
     priority = module.params["priority"]
@@ -380,6 +377,9 @@ def main():
             exception=traceback.format_exc(),
         )
 
+    if get_runtime_status():
+        semanage.semanage_set_reload(sh, 1)
+
     if state == "enabled":
         if path is not None:
             result = semodule_install(module, path, priority, sh)
```

With this patch, the role is able to install and uninstall modules also in SELinux disabled systems. I have test it only manually so there' might be some corner cases.

**Comment by martinpitt** [comment]
Discussed with @bachradsusi in chat. We don't actually need the set_reload(), as the role explicitly reloads later on (and the tests check that changes become effective). It would just unnecessarily slow down the role invocation.

Updated, let's hear the bots' opinion. Thanks!

**Review by spetrosi — APPROVED** [review]
